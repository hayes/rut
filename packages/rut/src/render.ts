import React from 'react';
import { act } from 'react-test-renderer';
import Renderer from './Renderer';
import { RendererOptions } from './types';

export default function render<Props>(
  element: React.ReactElement<Props>,
  options?: RendererOptions,
): Renderer<Props> {
  let renderer: Renderer<Props>;

  act(() => {
    renderer = new Renderer(element, options);
  });

  return renderer!;
}
