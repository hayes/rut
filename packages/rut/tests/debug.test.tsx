import React from 'react';
import render from '../src/render';

describe('debug()', () => {
  it('formats array props', () => {
    function ArrayProp({ list }: { list: unknown[] }) {
      return <ul />;
    }

    const wrapper = render(<ArrayProp list={['string', 123, true, null, {}, []]} />);

    expect(wrapper.debug()).toMatchSnapshot();
  });
});
