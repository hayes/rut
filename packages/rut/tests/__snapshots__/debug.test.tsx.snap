// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debug() formats array props 1`] = `
"<ArrayProp list={TODO}>
  <ul />
</ArrayProp>"
`;
